### PR TITLE
adds inflections for irregular plural crop names

### PR DESCRIPTION
**Pull request by brandonbaker40**
Adds inflections for irregular plural crop names. Issue #619 

- explicitly states plural form of all crops from [crop hierarchy](http://www.growstuff.org/crops/hierarchy) as of the date of this PR.

- BELOW is a list of crop name exceptions not yet included in this PR, but can be included if a contributor who knows the plural forms of these crops can help.

- pūhā
- bay
- rue
- brassica
- nettle
- Cucurbita maxima
- Cucurbita moschata
- Cucurbita pepo
- kawakawa
- karaka 
- hīnau
- tawa
- quince
- kūmara
- shiso
- pawpaw
- radicchio

/cc @Skud @Br3nda

### CODE REVIEW

**Comment by Br3nda** [comment]
Maybe this should be in the database instead. When adding a crop, also add the plural form of the noun in English.

**Comment by Br3nda** [comment]
For rhe words of Māori origin, when iswd in English their plural form is the same as English. Eg. One pūhā, twenty pūhā.

File: config/initializers/inflections.rb [hunk 1]
```diff
@@ -16,15 +16,202 @@
...
-  inflect.plural 'squash', 'squash'
-  inflect.plural 'bok choy', 'bok choy'
+
+  # crops
   inflect.plural 'achiote', 'achiote'
+  inflect.plural 'agave', 'agave'
   inflect.plural 'alfalfa', 'alfalfa'
   inflect.plural 'allspice', 'allspice'
-  inflect.plural 'spinach', 'spinach'
+  inflect.plural 'aloe vera', 'arugula'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: I dont think arugula matches with aloe vera

**Comment by Br3nda** [comment]
Thank you for working on this

**Comment by brandonbaker40** [comment]
@Br3nda I think adding plural forms to the database is a good long term solution. In the short term, is this enough to patch things up?

In a future release, we could....

**A) call a method that sets the plural form based on a value in the database, and only uses the Rails inflector if that value is `nil`** 

Example:
https://gist.github.com/brandonbaker40/be1150271dc62672d4e6bf30e2f34b3b

**OR**

**B) populate the entire database of crops with plural forms and make it a required field when creating a new crop**

C) something else, or a combination of A & B

/cc @Skud

**Comment by milesgould** [comment]
I'm pretty sure the plural of "nettle" is "nettles", and the plural of "brassica" is "brassicas" (Wiktionary agrees: [nettles](https://en.wiktionary.org/wiki/nettle), [brassicas](https://en.wiktionary.org/wiki/brassica)).

Also, people grow nettles deliberately? I knew you could eat them (and have done so myself), but I've always thought of them as a weed, and a painful one at that...

File: config/initializers/inflections.rb [hunk 2]
```diff
@@ -16,15 +16,207 @@
...
+  inflect.plural 'salad burnet', 'salad burnet'
+  inflect.plural 'salsify', 'salsify'
+  inflect.plural 'salvia', 'salvia'
+  inflect.plural 'sassafras', 'sassafras'
+  inflect.plural 'sesame', 'sesame'
+  inflect.plural 'shiitake', 'shiitake'
+  inflect.plural 'shiso', 'shiso'
+  inflect.plural 'smallage', 'smallage'
+  inflect.plural 'sorghum', 'sorghum'
+  inflect.plural 'soybean', 'soybean'
```
Comment (milesgould) [inline config/initializers/inflections.rb]: Shouldn't this be "soybeans"?

**Comment by milesgould** [comment]
The build's failing on a fairly trivial code-style error:
```
Errors on modified lines:
/home/travis/build/Growstuff/growstuff/spec/features/crops/inflections_spec.rb:58:1: C: Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body end.
```

File: config/initializers/inflections.rb [hunk 3]
```diff
@@ -16,15 +16,206 @@
...
+  inflect.plural 'cabbage', 'cabbage'
+  inflect.plural 'calamint', 'calamint'
+  inflect.plural 'cantueso', 'cantueso'
+  inflect.plural 'capsicum', 'capsica'
+  inflect.plural 'capsicum annuum', 'capsicum annuum'
+  inflect.plural 'Capsicum baccatum', 'Capsicum baccatum'
+  inflect.plural 'Capsicum chinense', 'Capsicum chinense'
+  inflect.plural 'cardinal vine', 'cardinal vine'
+  inflect.plural 'catnip', 'catnip'
+  inflect.plural 'cauliflower', 'cauliflower'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Cauliflowers

File: config/initializers/inflections.rb [hunk 4]
```diff
@@ -16,15 +16,206 @@
...
+  inflect.plural 'broccoflower', 'broccoflower'
+  inflect.plural 'broccoli', 'broccoli'
+  inflect.plural 'broccolini', 'broccolini'
+  inflect.plural 'Brussels sprouts', 'Brussels sprouts'
+  inflect.plural 'buckwheat', 'buckwheat'
+  inflect.plural 'burdock', 'burdock'
+  inflect.plural 'cabbage', 'cabbage'
+  inflect.plural 'calamint', 'calamint'
+  inflect.plural 'cantueso', 'cantueso'
+  inflect.plural 'capsicum', 'capsica'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Not sure if this. I think it should be capsicums.

File: config/initializers/inflections.rb [hunk 5]
```diff
@@ -16,15 +16,206 @@
...
-  inflect.plural 'spinach', 'spinach'
+  inflect.plural 'aloe vera', 'aloe vera'
+  inflect.plural 'amaranth', 'amaranth'
+  inflect.plural 'Amelanchier', 'Amelanchier'
+  inflect.plural 'anise', 'anise'
+  inflect.plural 'anise hyssop', 'anise hyssop'
+  inflect.plural 'arrowroot', 'arrowroot'
+  inflect.plural 'arugula', 'arugula'
+  inflect.plural 'asparagus', 'asparagus'
+  inflect.plural 'avocado', 'avocadoes'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Avocados

File: config/initializers/inflections.rb [hunk 6]
```diff
@@ -16,15 +16,206 @@
...
+  inflect.plural 'balm', 'balm'
+  inflect.plural 'bamboo', 'bamboo'
+  inflect.plural 'barley', 'barley'
+  inflect.plural 'basil', 'basil'
+  inflect.plural 'beebalm', 'beebalm'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Bee balms

File: config/initializers/inflections.rb [hunk 7]
```diff
@@ -16,15 +16,206 @@
...
+  inflect.plural 'Brassica rapa', 'Brassica rapa'
+  inflect.plural 'Brazilian starfish', 'Brazilian starfish'
+  inflect.plural 'breadfruit', 'breadfruit'
+  inflect.plural 'broccoflower', 'broccoflower'
+  inflect.plural 'broccoli', 'broccoli'
+  inflect.plural 'broccolini', 'broccolini'
+  inflect.plural 'Brussels sprouts', 'Brussels sprouts'
+  inflect.plural 'buckwheat', 'buckwheat'
+  inflect.plural 'burdock', 'burdock'
+  inflect.plural 'cabbage', 'cabbage'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Cabbages

File: config/initializers/inflections.rb [hunk 8]
```diff
@@ -16,15 +16,206 @@
...
+  inflect.plural 'anise', 'anise'
+  inflect.plural 'anise hyssop', 'anise hyssop'
+  inflect.plural 'arrowroot', 'arrowroot'
+  inflect.plural 'arugula', 'arugula'
+  inflect.plural 'asparagus', 'asparagus'
+  inflect.plural 'avocado', 'avocadoes'
+  inflect.plural 'balm', 'balm'
+  inflect.plural 'bamboo', 'bamboo'
+  inflect.plural 'barley', 'barley'
+  inflect.plural 'basil', 'basil'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Basils

File: config/initializers/inflections.rb [hunk 9]
```diff
@@ -16,15 +16,206 @@
...
+  inflect.plural 'buckwheat', 'buckwheat'
+  inflect.plural 'burdock', 'burdock'
+  inflect.plural 'cabbage', 'cabbage'
+  inflect.plural 'calamint', 'calamint'
+  inflect.plural 'cantueso', 'cantueso'
+  inflect.plural 'capsicum', 'capsica'
+  inflect.plural 'capsicum annuum', 'capsicum annuum'
+  inflect.plural 'Capsicum baccatum', 'Capsicum baccatum'
+  inflect.plural 'Capsicum chinense', 'Capsicum chinense'
+  inflect.plural 'cardinal vine', 'cardinal vine'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Vines

File: config/initializers/inflections.rb [hunk 10]
```diff
@@ -16,15 +16,206 @@
...
+  inflect.plural 'komatsuna', 'komatsuna'
+  inflect.plural 'kūmara', 'kūmara'
+  inflect.plural "lady's bedstraw", "lady's bedstraw"
+  inflect.plural 'lavender', 'lavender'
+  inflect.plural 'lemon myrtle', 'lemon myrtle'
+  inflect.plural 'lemonade', 'lemonade'
+  inflect.plural 'lemongrass', 'lemongrass'
+  inflect.plural 'lesser calamint', 'lesser calamint'
+  inflect.plural 'lesser galangal', 'lesser galangal'
+  inflect.plural 'lettuce', 'lettuce'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Lettuces

File: config/initializers/inflections.rb [hunk 11]
```diff
@@ -16,15 +16,206 @@
...
+  inflect.plural 'olluco', 'olluco'
+  inflect.plural 'oregano', 'oregano'
+  inflect.plural 'pandan', 'pandan'
+  inflect.plural 'parsley', 'parsley'
+  inflect.plural 'passion fruit', 'passion fruit'
+  inflect.plural 'pearl millet', 'pearl millet'
+  inflect.plural 'pennyroyal', 'pennyroyal'
+  inflect.plural 'peppermint', 'peppermint'
+  inflect.plural 'perilla', 'perilla'
+  inflect.plural 'persimmon', 'persimmon'
```
Comment (Br3nda) [inline config/initializers/inflections.rb]: Persimmons

**Pull request merged**